transport/transporters/s3/transporter/transporter.go

```diff
@@ -210,19 +212,15 @@ func (t *S3Transporter) transportWithRetry(ctx context.Context, messagesSlice []
 		return err, cancelled
 	}
 
-	gz.Reset(ioutil.Discard)
+	gz.Reset(io.Discard)
 
 	firstWalStart := messagesSlice[0].WalStart
 
-	// Free up messagesSlice now that we're done with it
-	messagesSlice = nil
```

Comment:
why are we no longer freeing this up?

Reply:
It technically wasn't necessary to begin with and only provided a hint to the GC.